pkg/plugin/manager.go

```diff
@@ -225,29 +225,84 @@ func (m *Manager) Install(url string) error {
 	pluginDirectory := filepath.Join(pluginsDir, pluginName)
 	pluginPath := filepath.Join(pluginDirectory, pluginName)
 
-	_, err = os.Stat(pluginDirectory)
+	if isNew {
+		_, err = os.Stat(pluginDirectory)
 
-	if os.IsNotExist(err) {
-		err := os.MkdirAll(pluginDirectory, os.ModePerm)
-		if err != nil {
-			return fmt.Errorf("creating %s plugin directory: creating folder %s: %w", archiveName, pluginDirectory, err)
-		}
-	} else {
-		if _, err = os.Stat(pluginPath); err == nil {
-			return fmt.Errorf("installing plugin in %s: Plugin Already Exists", pluginPath)
+		if os.IsNotExist(err) {
+			err := os.MkdirAll(pluginDirectory, os.ModePerm)
+			if err != nil {
+				return "", fmt.Errorf("creating plugin directory %s: %w", pluginDirectory, err)
+			}
+		} else if _, err = os.Stat(pluginPath); err == nil {
+			return "", fmt.Errorf("plugin %s already exists", pluginName)
 		}
 	}
 
 	err = unzip.Extract(resp.Filename, pluginDirectory)
 	if err != nil {
-		return fmt.Errorf("extracting the plugin at %s: %w", resp.Filename, err)
+		return "", fmt.Errorf("extracting the plugin at %s: %w", resp.Filename, err)
 	}
 
 	err = os.Rename(filepath.Join(pluginDirectory, pluginFilename), pluginPath)
+	if err != nil {
+		return "", fmt.Errorf("renaming plugin %s: %w", pluginName, err)
+	}
+
+	return pluginPath, nil
+}
+
+// Install grabs a remote plugin from the given url and install it locally.
+func (m *Manager) Install(url string) error {
+	pluginPath, err := m.DownloadPlugin(url, true)
+	if err != nil {
+		return fmt.Errorf("installing plugin at %s: %w", url, err)
+	}
 
 	err = m.InitPlugin(pluginPath)
 	if err != nil {
-		return fmt.Errorf("running plugin %s after installation: %w", pluginName, err)
+		return fmt.Errorf("running plugin %s after installation: %w", pluginPath, err)
+	}
+
+	return nil
+}
+
+func (m *Manager) Update(correlationID string) error {
+	pluginsDir := Directory()
+
+	plgn, err := m.Plugin(correlationID)
+	if err != nil {
+		return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
```

Comment:
you are not deleting the plugin here, you are getting the plugin from the node manager.